File: core/src/test/java/io/github/herrromich/batch/job/TestTask2.java
package io.github.herrromich.batch.job;

import io.github.herrromich.batch.TaskPriorities;
import org.jetbrains.annotations.NotNull;

import java.util.Set;

public class TestTask2 extends TestTask {
    @NotNull
    @Override
    public String getName() {
        return "test-task-2";
    }

    @Override
    public int getPriority() {
        return TaskPriorities.HIGHER;
    }

    @NotNull
    @Override
    public Set<String> getProducibles() {
        return Set.of("test-resource-1");
    }
}

File: builder/src/test/kotlin/io/github/herrromich/batch/TestTask2.kt
package io.github.herrromich.batch

class TestTask2 : TestTask("test-task2") {
    override val priority = TaskPriorities.CRITICAL
    override val failLevel = FailLevel.FATAL
    override val consumables = setOf("test-resource1", "test-resource2")
    override val producibles = setOf("test-resource2")
}